### PR TITLE
models changes and brandroute

### PR DESCRIPTION
**Pull request by zhala07**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "multer": "^1.4.5-lts.1",
     "mysql2": "^3.7.0",
     "sequelize": "^6.35.2",
+    "ts-node": "^10.9.2",
```
Comment (Yousefjb) [inline package.json:22]: why adding ts-node if you are not using it ?
Also it is better to be in devDependencies section. The same for typescript package

File: src/config/database.ts [hunk 2]
```diff
@@ -3,13 +3,15 @@ import dotenv from "dotenv";
...
-    "eCommerceTap",
-    process.env.DB_USERNAME as string,
-    process.env.DB_PASSWORD,
-    {
-      host: process.env.DB_HOSTNAME,
-      dialect: process.env.DB_DIALECT as 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql',
-      port: Number(process.env.DB_PORT),
-    }
-  );
+  "ecommercetap",
```
Comment (Yousefjb) [inline src/config/database.ts:6]: database name could be also in env variable like the others

File: src/controllers/productsController.ts [hunk 3]
```diff
@@ -12,107 +12,125 @@ dotenv.config();
...
+    let page = Number(req.query.page) || 1;
+    let limit = Number(req.query.limit) || 9;
 
-  if (page < 0) {
-    page = 1;
-  }
-  if (limit > 80 || limit < 0) {
-    limit = 9;
-  }
+    if (page < 0) {
```
Comment (Yousefjb) [inline src/controllers/productsController.ts:21]: you could use Math.max( ... , 1)

File: src/controllers/productsController.ts [hunk 4]
```diff
@@ -12,107 +12,125 @@ dotenv.config();
...
-        ...conditions["price"],
-        [Op.lte]: lessThan,
-      },
-    };
-  }
+    if (lessThan) {
+      conditions = {
+        ...conditions,
+        price: {
+          ...conditions['price'],
```
Comment (Yousefjb) [inline src/controllers/productsController.ts:60]: What is the purpose of this ? and why it is not used in the greaterThan block ?

File: src/controllers/productsController.ts [hunk 5]
```diff
@@ -12,107 +12,125 @@ dotenv.config();
...
+      sort = [[`${sort.replace('-', '')}`, dir]];
+    } else {
+      sort = [['id', 'ASC']];
+    }
+
+    const category = req.query.category as unknown as String;
+
+    if (
+      category &&
+      (category == 'Steel' ||
```
Comment (Yousefjb) [inline src/controllers/productsController.ts:82]: it would be better to define this list of categories in constant so you could reuse it later and give it a name so we understand what these categories are

File: src/models/brands.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    id: {
+        type: DataTypes.INTEGER,
+        autoIncrement: true,
+        primaryKey: true,
+      },
+      name: {
+        type:DataTypes.STRING(40),
+        allowNull:false,
+      },
+      image_secure_url:{
```
Comment (Yousefjb) [inline src/models/brands.ts:13]: Why did you add "secure" ? is the url encrypted for some reason ?

File: src/models/cart.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ export const CartsModel = sequelize.define('cart', {
     autoIncrement: true,
     primaryKey: true,
   },
-  product_name: {
+  product_name: { // should be id not name cuz its a FK
```
Comment (Yousefjb) [inline src/models/cart.ts:11]: if it is something you intend to do you can add `TODO:` to clarify your intention, otherwise you communicate this error to your team in slack

File: src/config/database.ts (hunk 2 above)
Comment (zhala07) [reply src/config/database.ts:6]: It's how Mohammad did it, and I didn't change it.

File: package.json (hunk 1 above)
Comment (zhala07) [reply package.json:22]: I added it during debugging  and forgot to remove it

File: src/models/products.ts [hunk 8]
```diff
@@ -49,6 +49,13 @@ export const ProductsModel = sequelize.define("products", {
     type: DataTypes.STRING(128),
     allowNull: true,
   },
+  brand_id :{   // add it
```
Comment (Yousefjb) [inline src/models/products.ts:52]: don't add unnecessary comments

File: src/controllers/productsController.ts (hunk 3 above)
Comment (zhala07) [reply src/controllers/productsController.ts:21]: This was done by Mohammad.

File: src/config/database.ts [hunk 9]
```diff
@@ -3,13 +3,15 @@ import dotenv from "dotenv";
...
+  process.env.DB_USERNAME as string,
+  process.env.DB_PASSWORD,
+  {
+    host: process.env.DB_HOSTNAME,
+    dialect:"mysql",
```
Comment (Yousefjb) [inline src/config/database.ts:11]: very good, dialect should not be a variable

File: src/controllers/productsController.ts (hunk 4 above)
Comment (zhala07) [reply src/controllers/productsController.ts:60]: this one also was done by Mohammad

File: src/controllers/brandsContraller.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
+// brandsController.ts
+import { Request, Response } from 'express';
+import { BrandsModel } from '../models/brands';
+
+
+export const getALLBrands = async (req:Request , res:Response) =>{
+    try {
+       
+        const Brands = await BrandsModel.findAll();
+        return res.status(200).json({ data: { message:"success" , Brands} })
```
Comment (Yousefjb) [inline src/controllers/brandsContraller.ts:10]: no need for message: "success" in the response, it adds no value to the client.
response could be simplified as  `.json(Brands)`

File: src/models/cart.ts (hunk 7 above)
Comment (zhala07) [reply src/models/cart.ts:11]: "I did that on Slack, and I also left a comment so they would notice it."

File: src/config/database.ts (hunk 9 above)
Comment (zhala07) [reply src/config/database.ts:11]: This is the only thing I added to this, so thank you.

File: src/models/products.ts (hunk 8 above)
Comment (zhala07) [reply src/models/products.ts:52]: Just to ensure they notice it, then it can be removed.

File: src/controllers/brandsContraller.ts (hunk 10 above)
Comment (zhala07) [reply src/controllers/brandsContraller.ts:10]: I will update it

File: src/models/brands.ts (hunk 6 above)
Comment (zhala07) [reply src/models/brands.ts:13]: No, I kept it in Mohammad's way because he named it that way in other models.

File: src/controllers/productsController.ts (hunk 5 above)
Comment (zhala07) [reply src/controllers/productsController.ts:82]: We will

Comment (MohammadAhmadKhader) [reply src/controllers/productsController.ts:82]: the categories was not agreed on before we had to add ones for testing, but we did recently and will do a complete implementation for it.

File: src/models/brands.ts (hunk 6 above)
Comment (MohammadAhmadKhader) [reply src/models/brands.ts:13]: secure_url as it named by the Cloudinary, its a link to the image in the fileSystem.

File: src/controllers/productsController.ts (hunk 4 above)
Comment (MohammadAhmadKhader) [reply src/controllers/productsController.ts:60]: The purpose is in case the user wants to choose greatThan and lessThan at same time to have them both, so he select between (10 and 200) and etc , if he wants (price > 100) he also can do it, if he wants (price < 100) he also can do it, with this implementation.
